ide/idetop.ml

```diff
@@ -544,16 +547,40 @@ let rec parse = function
        x :: parse rest
   | [] -> []
 
-let () = Usage.add_to_usage "coqidetop"
-"  --xml_format=Ppcmds    serialize pretty printing messages using the std_ppcmds format\
+let coqidetop_specific_usage = Usage.{
+  executable_name = "coqidetop";
+  extra_args = "";
+  extra_options = "\n\
+coqidetop specific options:\n\
+\n  --xml_formatinclude dir           (idem)\
+\n  --xml_format=Ppcmds    serialize pretty printing messages using the std_ppcmds format\
 \n  --help-XML-protocol    print documentation of the Coq XML protocol\n"
+}
 
-let islave_init ~opts extra_args =
-  let args = parse extra_args in
-  CoqworkmgrApi.(init High);
-  opts, args
+let islave_parse ~opts extra_args =
+  let open Coqtop in
+  let run_mode, extra_args = coqtop_toplevel.parse_extra ~opts extra_args in
+  let extra_args = parse extra_args in
+  (* Pass extra arguments to coqide via stdout *)
```

Comment:
@herbelin can you add a comment / clarify how stdout is involved here? It is not obvious to me what is going on, but I am not expert on IDE code.

Using stdout in the parse function seems a bit strange to me.

Reply:
Ping @herbelin

Reply:
I'll look at that within a few days.

Reply:
@ejgallego: I added the following comments:
```
+  (* One of the role of coqidetop is to find the name of buffers to open *)
+  (* in the command line; Coqide is waiting these names on stdout *)
+  (* (see filter_coq_opts in coq.ml), so we send them now *)
```
I'm not saying that this is a good way to do. Maybe the names of buffers to open should be passed instead on a special communication channel?

Reply:
Thanks for the info @herbelin , let me have a look.

Reply:
Indeed I understand the mechanism now, thanks, I was not aware of it. It seems better the way you propose.